src/main/java/at/hannibal2/skyhanni/features/combat/BroodmotherFeatures.kt

```diff
@@ -0,0 +1,118 @@
+            lastStage = currentStage
+            currentStage = newStage
+            onStageUpdate()
+        }
+    }
+
+    private fun onStageUpdate() {
+        val currentStageEntry = StageEntry.valueOf(currentStage.replace("!", "").uppercase())
+        currentStage = currentStageEntry.toString()
+        ChatUtils.debug("New Broodmother stage: $currentStage")
+
+        if (lastStage == "" && currentStageEntry != StageEntry.ALIVE) {
+            if (config.stageOnJoin) {
+                ChatUtils.chat("The Broodmother's current stage in this server is $currentStage§r.")
+            }
+            return
+        }
+        // ignore if the stage temporarily reverts to Imminent after the Broodmother's death
+        if (currentStageEntry == StageEntry.IMMINENT && lastStage == StageEntry.ALIVE.toString()) {
+            return
+        }
+
+        if (currentStageEntry == StageEntry.ALIVE) {
+            broodmotherSpawnTime = SimpleTimeMark.farPast()
+            ChatUtils.clickableChat("The Broodmother has spawned! Click here to warp to the Spider Mound!", onClick = { HypixelCommands.warp("nest")})
+            if (config.spawnAlertEnabled) {
+                val alertSound = SoundUtils.createSound(config.spawnAlert.alertSound, config.spawnAlert.pitch)
+                SoundUtils.repeatSound(100, config.spawnAlert.repeatSound, alertSound)
+                LorenzUtils.sendTitle(config.spawnAlert.text.replace("&", "§"), 3.seconds)
+            }
+            return
+        }
+
+        val timeUntilSpawn  = currentStageEntry.minutes.minutes
+        broodmotherSpawnTime = SimpleTimeMark.now() + timeUntilSpawn
+
+        if (currentStageEntry == StageEntry.IMMINENT && config.imminentWarningEnabled) {
+            SoundUtils.repeatSound(100, 2, SoundUtils.createSound("note.pling", 0.5f))
+            ChatUtils.chat("The Broodmother is §4Imminent§e! It will spawn in §b60 seconds§e!")
+            return
+        }
+
+        val matchStageEntry = config.stages.find { it.toString() == currentStage }
+        if (matchStageEntry != null) {
+
+            if (currentStageEntry == StageEntry.SLAIN && !(config.hideSlainWhenNearby && LorenzUtils.skyBlockArea == "Spider Mound")) {
+                ChatUtils.chat("The Broodmother was killed!")
+            }
+            else if (currentStageEntry != StageEntry.SLAIN) {
+                ChatUtils.chat("Broodmother: $lastStage §e-> $currentStage§e. §b${timeUntilSpawn.inWholeMinutes} minutes §euntil it spawns!")
+            } else {
+                broodmotherSpawnTime = SimpleTimeMark.now() + 10.minutes
+            }
+        }
+    }
+
+    @SubscribeEvent
+    fun onWorldChange(event: LorenzWorldChangeEvent) {
+        broodmotherSpawnTime = SimpleTimeMark.farPast()
+        lastStage = ""
+        currentStage = ""
+    }
+
+    @SubscribeEvent
+    fun onRenderOverlay(event: GuiRenderEvent.GuiOverlayRenderEvent) {
+        if (!config.countdownEnabled) return
```

Comment:
please always check for "is in skyblock" before accessing the config, especially in events that maybe get fired outside a minecraft world